### ISSUE TITLE
Fail to incarnate with normal port and multiplicity port when connected in reverse order

### ISSUE THREAD

**Issue by MartinOrell** [secure-dev-ops/code-realtime#20]
It seems when you have both a normal port and a multiplicity port they need to be connected in the same order as they are defined in the capsule.
I have a capsule *A* with 2 ports
```
service behavior port cPort: Protocol[2];
service behavior port bPort : Protocol;
```
I connect them in my top capsule to ports on other capsules
```
connect a.bPort with b.aPort;
connect a.cPort with c.aPort;
```
When trying to run the program I get this message:
```
incarnate(application(Top)<<machine>>.a[0]): Insufficient interface replication.
```

There are multiple ways I could get rid of this error:
If I change the order the ports were defined
If I change the order the connections were made
If I changed the multiplicity port to a normal port
If I changed the normal port to be a multiplicity port.

For example, changing the order like this fixed the error:
```
service behavior port bPort : Protocol;
service behavior port cPort: Protocol[2];
```
It would also be helpful if the message could be made more specific why the incarnation failed. "Insufficient interface replication" does not explain to me much what went wrong or that there were anything wrong with the ports.

My whole test program:
```
protocol Protocol{
    in message();
};

capsule CapsuleA{

    service behavior port cPort: Protocol[2];
    service behavior port bPort : Protocol;
    
    statemachine{
        state S;
        initial -> S;
    };
};

capsule CapsuleB{

    service behavior port aPort~: Protocol;

    statemachine{
        state S;
        initial -> S;
    };
};

capsule CapsuleC{

    service behavior port aPort~: Protocol;

    statemachine{
        state S;
        initial -> S;
    };
};

capsule Top{

    part a : CapsuleA;
    part b : CapsuleB;
    part c : CapsuleC[2];

    connect a.bPort with b.aPort;
    connect a.cPort with c.aPort;

    statemachine{
        state S;
        initial -> S;
    };
};
```

**Comment by mattias-mohlin** [comment]
We were able to reproduce this problem, and thanks to your comments about the significance of port order, we think we know the reason for the failure (elements of rtg_interface arrays in generated code get sorted alphabetically according to port name, and the port declaration order is not respected). We need to analyze a bit more, but hope we can include a fix with version 2.0 which is expected to be released in approx 2 weeks from now.

**Comment by mattias-mohlin** [comment]
This scenario should now work in Code RealTime 2.0. However, we have noticed that if you have a port on the capsule that is not connected, you can still run into this problem (again it will depend on the order of the unconnected port). Unconnected ports are reported by a warning, but those warnings can be ignored if you plan to use the Art Debugger for sending events on the unconnected port. Therefore we should also support this scenario (planned for 2.0.1).
